web/src/instrumentedComponents.tsx

```diff
@@ -27,3 +39,33 @@ export function InstrumentedButton(props: InstrumentedButtonProps) {
     />
   )
 }
+
+export function InstrumentedTextField(
+  props: TextFieldProps & InstrumentationProps
+) {
+  const { analyticsName, analyticsTags, onChange, ...textFieldProps } = {
+    ...props,
+  }
+
+  // we have to memoize the debounced function so that incrs reuse the same debounce timer
+  const debouncedIncr = useMemo(
+    () =>
+      // debounce so we don't send analytics for every single keypress
+      debounce(() => {
+        incr(analyticsName, {
+          action: AnalyticsAction.Edit,
+          ...(analyticsTags ?? {}),
+        })
+      }, 5000),
```

Comment:
(optional suggestion) it might be nice to extract this time into a `const` for readability and for use in the tests.

also, five seconds seems a little long to wait for text input to change. i think in other places, we're using half a second?

Reply:
> it might be nice to extract this time into a const for readability and for use in the tests.

yeah, fair

> also, five seconds seems a little long to wait for text input to change. i think in other places, we're using half a second?

Those are for the purposes of re-rendering / applying filters, yeah? I think it makes sense to have dramatically different debounces for those.
e.g., if the user spends 5 seconds typing in a new filter value:
* we want to incrementally update the filtered results as they type, so they can see the impact of what they've typed and judge whether they should type more to further narrow results
* the analytics are invisible to the user and logging four different "change" events doesn't seem particularly useful

Reply:
hm, or maybe I misunderstood the comment?

Reply:
ah, yeah! that's a good point. i forgot about the debounce being used both to apply the filter and log the event. it makes sense to keep the filter term field logic on its own. then the goal of the five second delay here is to attempt to log only one event per input interaction, or rather to prevent extraneous multiple events from a single interaction.